core/src/main/java/hudson/model/FreeStyleProject.java

```diff
@@ -72,5 +74,25 @@ public String getDisplayName() {
         public FreeStyleProject newInstance(ItemGroup parent, String name) {
             return new FreeStyleProject(parent,name);
         }
+
+        @Override
+        public String getDescription() {
+            return Messages.FreeStyleProject_Description();
+        }
+
+        @Override
+        public String getCategoryId() {
+            return StandaloneProjectsCategory.ID;
+        }
+
+        @Override
+        public String getIconFilePathPattern() {
+            if (Jenkins.RESOURCE_PATH.isEmpty()) {
+                return "images/:size/freestyleproject.png";
+            } else {
+                return Jenkins.RESOURCE_PATH.substring(1) + "/images/:size/freestyleproject.png";
```

Comment:
:ant: e3b11aefa28a524fe739ce4361af7fba55fa858e did not really explain anything, but the `if` switch seems like overkill. The original code

```java
return Jenkins.RESOURCE_PATH + "images/:size/freestyleproject.png";
```

presumably did not work since it returned something like

    /static/abc123images/:size/freestyleproject.png

So it is clearer to fix that using e.g.

```java
return (Jenkins.RESOURCE_PATH + "/images/:size/freestyleproject.png").replaceFirst("^/", "");
```

or simply make the caller of `getIconFilePathPattern()` tolerate a leading slash.